sequence-generator/src/main/java/org/terracotta/management/sequence/Defaults.java

```diff
@@ -0,0 +1,163 @@
+    public long getTimestamp() {
+      return delegate.getTimestamp();
+    }
+  };
+
+  private static final Comparator<NetworkInterface> NETWORK_INTERFACE_COMPARATOR = new Comparator<NetworkInterface>() {
+    @Override
+    public int compare(NetworkInterface o1, NetworkInterface o2) {
+      return o1.getName().compareTo(o2.getName());
+    }
+  };
+
+  static byte[] readFirstNonLoopbackMacAddress() {
+    List<NetworkInterface> networkInterfaces;
+    try {
+      networkInterfaces = Collections.list(NetworkInterface.getNetworkInterfaces());
+    } catch (SocketException e) {
+      throw new IllegalStateException("Machine has no network interfaces!", e);
+    }
+
+    // order interfaces by name
+    Collections.sort(networkInterfaces, NETWORK_INTERFACE_COMPARATOR);
+
+    // find the first accessible non-loopback interface having a mac address
+    // we try first to skip all virtual interfaces since they can be easily dynamically created on-demand
+    // the goal of this method is to try return the same value each time as possible
+    // we consider only "parent" interfaces
+    for (NetworkInterface networkInterface : networkInterfaces) {
+      try {
+        byte[] mac;
+        if (!networkInterface.isLoopback()
+            && !networkInterface.isPointToPoint()
+            && !networkInterface.isVirtual()
+            && networkInterface.getParent() == null
+            && (mac = networkInterface.getHardwareAddress()) != null
+            && mac.length >= 6) {
+          return mac;
+        }
+      } catch (SocketException ignored) {
+      }
+    }
+
+    // if we do not succeed, we enlarge our search
+    for (NetworkInterface networkInterface : networkInterfaces) {
+      try {
+        byte[] mac;
+        if (!networkInterface.isLoopback()
+            && (mac = networkInterface.getHardwareAddress()) != null
+            && mac.length >= 6) {
+          return mac;
+        }
+      } catch (SocketException ignored) {
+      }
+    }
+
+    throw new IllegalStateException("Unable to read a MAC address");
+  }
+
+  static long readPID() {
+    String name = ManagementFactory.getRuntimeMXBean().getName();
+    long pid = 0;
+    for (int i = 0; i < name.length() && Character.isDigit(name.charAt(i)); i++) {
+      pid = pid * 10 + Character.getNumericValue(name.charAt(i));
+    }
+    return pid;
+  }
+
+  private static <T> T findBest(Class<T> type, T def) {
+    // try first a JVM sysprop
+    String cName = System.getProperty(type.getName());
+    if (cName != null) {
+      try {
+        return type.cast(Defaults.class.getClassLoader().loadClass(cName).newInstance());
+      } catch (Exception e) {
+        throw new IllegalArgumentException("Unable to instance  " + type.getSimpleName() + " " + cName + " set from system property");
+      }
+    } else {
+      // otherwise, try a service implementation
+      try {
+        return ServiceLoader.load(type).iterator().next();
+      } catch (Exception ignored) {
```

Comment:
Because its is the "Best we find" strategy: its tries to get from the sysprop, if not found then tries from a service, otherwise stick to the default (mac_pid or system time).